src/app/nodetool/qgsmaptoolnodetool.cpp

```diff
@@ -186,14 +200,23 @@ void QgsMapToolNodeTool::canvasPressEvent( QgsMapMouseEvent* e )
 
     if ( snapResults.size() < 1 )
     {
-      emit messageEmitted( tr( "could not snap to a segment on the current layer." ) );
-      return;
+      QgsFeature feature = getFeatureAtPoint( e );
+      if ( feature.geometry() == NULL ) {
+	emit messageEmitted(tr( "could not snap to a segment on the current layer." ) );
```

Comment:
People facing this message don't sometimes know what they should do. It may be worth adding a more explicite message to tell user what may be the reason of the fail. Sthg like : "could not snap to a segment or a vertex on the current layer. Please click closer to the feature or enhance the search tolerance under Settings->Options->Digitizing->Search radius for vertex edits."